src/fondant/pipeline/pipeline.py

```diff
@@ -185,25 +184,59 @@ def __init__(
         self.resources = resources or Resources()
 
     @classmethod
-    def from_component_yaml(cls, path, **kwargs) -> "ComponentOp":
+    def from_component_yaml(cls, path, fields=None, **kwargs) -> "ComponentOp":
         if cls._is_custom_component(path):
             component_dir = Path(path)
         else:
             component_dir = cls._get_registry_path(str(path))
+
         component_spec = ComponentSpec.from_file(
             component_dir / cls.COMPONENT_SPEC_NAME,
         )
 
+        # If consumes is not defined in the pipeline, we will try to infer it
+        if kwargs.get("consumes", None) is None:
```

Comment:
```suggestion
        if kwargs.get("consumes") is None:
```

`None` is the default :slightly_smiling_face: